programs/bpf_loader/src/lib.rs

```diff
@@ -495,6 +495,82 @@ fn process_instruction_common(
     }
 }
 
+fn process_set_authority(
+    instruction_context: &InstructionContext,
+    transaction_context: &TransactionContext,
+    log_collector: &Option<Rc<RefCell<LogCollector>>>,
+    new_authority_must_sign: bool,
+) -> Result<(), InstructionError> {
+    instruction_context.check_number_of_instruction_accounts(2)?;
+    let mut account = instruction_context.try_borrow_instruction_account(transaction_context, 0)?;
+    let present_authority_key = transaction_context.get_key_of_account_at_index(
+        instruction_context.get_index_of_instruction_account_in_transaction(1)?,
+    )?;
+    let new_authority = instruction_context
+        .get_index_of_instruction_account_in_transaction(2)
+        .and_then(|index_in_transaction| {
+            transaction_context.get_key_of_account_at_index(index_in_transaction)
+        })
+        .ok();
+
+    match account.get_state()? {
+        UpgradeableLoaderState::Buffer { authority_address } => {
```

Comment:
I think it would make sense to apply `new_authority_must_sign` to the buffer authority as well.

Reply:
done and added a test for set_buffer_authority_checked